api/src/main/java/org/openmrs/util/OpenmrsUtil.java

```diff
@@ -812,39 +812,9 @@ public static int compareWithNullAsLatest(Date d1, Date d2) {
 		return ret;
 	}
 	
-	// TODO: properly handle duplicates
 	public static List<Concept> conceptListHelper(String descriptor) {
 		List<Concept> ret = new ArrayList<>();
-		if (descriptor == null || descriptor.length() == 0) {
-			return ret;
-		}
-		ConceptService cs = Context.getConceptService();
-		
-		for (StringTokenizer st = new StringTokenizer(descriptor, "|"); st.hasMoreTokens();) {
-			String s = st.nextToken().trim();
-			boolean isSet = s.startsWith("set:");
-			if (isSet) {
-				s = s.substring(4).trim();
-			}
-			Concept c = null;
-			if (s.startsWith("name:")) {
-				String name = s.substring(5).trim();
-				c = cs.getConceptByName(name);
-			} else {
-				try {
-					c = cs.getConcept(Integer.valueOf(s.trim()));
-				}
-				catch (Exception ex) {}
-			}
-			if (c != null) {
-				if (isSet) {
-					List<Concept> inSet = cs.getConceptsByConceptSet(c);
-					ret.addAll(inSet);
-				} else {
-					ret.add(c);
-				}
-			}
-		}
+		ret.addAll(conceptSetHelper(descriptor));
```

Comment:
You basically deprecated `conceptListHelper(String)` saying that it would just do what `conceptSetHelper(String)` does. So before we do that, we need to figure out what the original intent of both methods was. And then **add the appropriate JavaDoc**.

@dkayiwa do you remember by any chance what does methods were for?